connectors/aws/aws-comprehend/src/main/java/io/camunda/connector/comprehend/model/ComprehendRequestDataDeserializer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ *       under one or more contributor license agreements. Licensed under a proprietary license.
+ *       See the License.txt file for more information. You may not use this file
+ *       except in compliance with the proprietary license.
+ */
+package io.camunda.connector.comprehend.model;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+
+public class ComprehendRequestDataDeserializer extends JsonDeserializer<ComprehendRequestData> {
+
+  @Override
+  public ComprehendRequestData deserialize(JsonParser jp, DeserializationContext ctxt)
+      throws IOException, JacksonException {
+    JsonNode node = jp.getCodec().readTree(jp);
+
+    var operationType = determineActionType(node);
+    ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+
+    if (operationType.equals("sync")) {
+      return mapper.convertValue(node, ComprehendSyncRequestData.class);
+    }
+    if (operationType.equals("async")) {
+      return mapper.convertValue(node, ComprehendAsyncRequestData.class);
+    }
+    throw new UnsupportedOperationException("Unsupported execution type: " + operationType);
+  }
```

Comment:
This should work by just using jackson annotations. You can use a default mapping in jackson.

Reply:
done